src/v3/index.ts

```diff
@@ -0,0 +1,37 @@
+import { PluginEventExtra } from '../plugin'
+import { userAgentV2 } from '../v2'
+import { detectBrowser, detectBrowserVersion, detectDevice, detectDeviceType, detectOS } from './user-agent-utils'
+
+export const userAgentV3 = (
+    event: PluginEventExtra,
+    userAgent: string,
+    global: {
+        enabledPlugin: boolean
+        enableSegmentAnalyticsJs: boolean
+        overrideUserAgentDetails: boolean
+        debugMode: boolean
+    }
+): PluginEventExtra => {
+    if (event.properties['$user_agent_plugin_disable']) {
+        return event
+    }
+
+    const vendor = event.properties['$navigator_vendor']
+
+    event.properties['$device'] = detectDevice(userAgent)
```

Comment:
I think we use `$device_name` and/or `$device_name` on Mobile, I wonder if we should unify that? otherwise the filtering experience isn't ideal unless the filtering considers all those variations?